### ISSUE TITLE
fact acceptor introduced warnings

### ISSUE THREAD

**Issue by ashermancinelli** [LLNL/hiop#163]
After fact acceptor PR #149 was merged, I'm seeing new warnings ([in this pipeline for example](https://gitlab.pnnl.gov/exasgd/solvers/hiop-mirror/-/jobs/19613)):

```console
/people/svcexasgd/gitlab/19613/src/Optimization/hiopFactAcceptor.hpp(105): warning: invalid narrowing conversion from "long long" to "int"
/people/svcexasgd/gitlab/19613/src/Optimization/hiopKKTLinSys.hpp(319): warning: missing return statement at end of non-void function "hiop::hiopKKTLinSysLowRank::updateMatrix"
```

From my understanding, `long long` -> `int` conversions may have been causing trouble with copyRowsFrom kernel (eg in #103).

**Comment by ashermancinelli** [comment]
@nychiang, do you mind if I assign this to you since it seems to have come from your PR?

**Comment by nychiang** [comment]
Sure. Please assign it to me.
The first warning can be easily fixed. 
We just need to change ` n_required_neg_eig_{n_required_neg_eig}` to ` n_required_neg_eig_(n_required_neg_eig)`, since list initialization does not allow narrowing conversion.
However, this doesn't solve the problem that we need to narrowing convert `long long` to `int`, and HiOp uses this conversion in many places, e.g., [here](https://github.com/LLNL/hiop/blob/7e8adae9db757aed48e5c2bc448316307598258f/src/Optimization/hiopKKTLinSysMDS.cpp#L101)
The main issue here is that all the user input matrices use `long long`, but in the linear algebra level all these values are converted to `int`. I discussed this narrowing conversion issue with Cosmin long time ago, and we agreed that this is not an urgent issue and they are safe right now.

**Comment by nychiang** [comment]
fixed in #164
